src/runtime/virtcontainers/nydusd.go

```diff
@@ -0,0 +1,424 @@
+package virtcontainers
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/kata-containers/kata-containers/src/runtime/pkg/katautils/katatrace"
+	"github.com/kata-containers/kata-containers/src/runtime/virtcontainers/utils/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	infoEndpoint  = "/api/v1/daemon"
+	mountEndpoint = "/api/v1/mount"
+
+	defaultHttpClientTimeout = 30 * time.Second
+	contentType              = "application/json"
+
+	maxIdleConns          = 10
```

Comment:
What's the impact of each idle connection? ooi, how did you calculate this default value?

Reply:
The number of connections between kata-shim-v2 with nydusd is small, and the default value is  a experience value.